src/buffer.rs

```diff
@@ -243,6 +248,19 @@ impl Buffer {
         Ok(())
     }
 
+    /// Takes other buffer, measures its length and prints this buffer from the point where
+    /// the other stopped.
+    /// Used to implement autosuggestions.
+    pub fn print_rest<W>(&self, out: &mut W, other: &Buffer) -> io::Result<usize>
+        where W: Write
+    {
+        let matching_part = other.data.len();
+        let string: String = self.data.iter().skip(matching_part).cloned().collect();
+        out.write(string.as_bytes())?;
```

Comment:
@msehnout If you need to, you can calculate the size of a character with the `len_utf8()` method to get the correct index value to slice by bytes.